### ISSUE TITLE
(aws-codeguruprofiler): Add CodeGuru Lambda Layer Resource

### ISSUE THREAD

**Issue by jonnekaunisto** [aws/aws-cdk#13001]
I am proposing adding a Resource for CodeGuru Lambda Layer. Currently when setting up Code Guru layer you will need to do something like this:
```
import lambda = require('@aws-cdk/aws-lambda');
const layerArn = `arn:aws:lambda:us-west-1:580247275435:layer:AWSCodeGuruProfilerJavaAgentLayer:3`;
const layer = lambda.LayerVersion.fromLayerVersionArn(this, `LayerFromArn`, layerArn);
```

I propose it should be more like:
```
new CodeGuruLayer(this, "CodeGuruLayer', {});
```

### Use Case
1. We wouldn't need to hardcode arns anymore which leaves room for error
2. The resource could by default use the latest version so that consumers would not need to keep track of what version they are using(we should still provide option to use any version through props)
3. We can add the supported runtimes to the resource so that issues can be found at build time
4. We can also add supported regions to the resource, to find issues at build time

### Proposed Solution

Proposed solution could roughly look like this. This would require exporting [LayerVersionBase](https://github.com/aws/aws-cdk/blob/5664480a97958263ee7cb903c2aff0276e738dc3/packages/%40aws-cdk/aws-lambda/lib/layers.ts#L88) from lambda module. Or maybe another abstract class is warranted. That abstract class could be used for other layers as well, such as the lambda insights layer.

```
export interface CodeGuruLayerProps {
  readonly region: string;
  readonly version?: number;
  readonly compatibleRuntimes?: Runtime[];
}

export class CodeGuruLayer extends LayerVersionBase {
  public static LATEST_VERSION = 3; // eslint-disable-line
  public readonly layerVersionArn: string;
  public readonly compatibleRuntimes?: Runtime[] = Runtime.ALL;

  public constructor(scope: Construct, id: string, props: CodeGuruLayerProps) {
    super(scope, id);

    const version = props.version != undefined ? props.version : CodeGuruLayer.LATEST_VERSION;

    if (version > CodeGuruLayer.LATEST_VERSION || version < 1) {
      throw new Error(`Not a valid CodeGuru version: ${version}`);
    }

    this.layerVersionArn = `arn:aws:lambda:${props.region}:157417159150:layer:AWSCodeGuruProfilerJavaAgentLayer:${version}`;
  }
}
```

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by nija-at** [comment]
I would propose that we go one step further. If the runtime is Python and the the `profiling` property is set, we should just add the layer to the function.

@jonnekaunisto - would that satisfy your requirement as well?

**Comment by nija-at** [comment]
Having read this further - https://docs.aws.amazon.com/codeguru/latest/profiler-ug/python-lambda-layers.html - we may need a few more options to correctly configure all of this.

**Comment by jonnekaunisto** [comment]
Yeah setting the layer when profiling flag is set seems like a good idea

**Comment by sameergoyal** [comment]
This might also be useful for JVM Lambdas on Java 8 Coretto or Java 11 which can do with a similar setup for lambda layers: https://docs.aws.amazon.com/codeguru/latest/profiler-ug/lambda-simple.html instead of the more involved setup (with code changes) needed for other JVM runtimes.

**Comment by greg5123334** [comment]
I seem to be experiencing a similar problem. CDK provisions the resources and configurations as per the [prerequisites](https://docs.aws.amazon.com/codeguru/latest/profiler-ug/python-lambda-layers.html), and after confirming that the layer, env vars, and permissions are in place, the ProfilingGroup still remains in a _Setup Required_ state. For reference, please find a snippet of my code below:

**CDK**
```
        let region = Stack.of(this).region
        const CodeGuruLayer = LayerVersion.fromLayerVersionArn(this, "CodeGuruLayer",
            'arn:aws:lambda:' + region + ':157417159150:layer:AWSCodeGuruProfilerPythonAgentLambdaLayer:11',
        )

        const fn = new lambda.Function(this, 'MyFunction', {
            runtime: lambda.Runtime.PYTHON_3_9,
            handler: 'base.main',
            code: Code.fromAsset("./assets"),
            profiling: true,
            environment: {
                'AWS_LAMBDA_EXEC_WRAPPER': '/opt/codeguru_profiler_lambda_exec',
                'AWS_CODEGURU_PROFILER_TARGET_REGION': region,
            },
            layers: [CodeGuruLayer],
        });
        fn.role?.addManagedPolicy(
            iam.ManagedPolicy.fromAwsManagedPolicyName(
                'AmazonCodeGuruProfilerAgentAccess',
            ),
        );
```

**lambda handler: base.py**
```
import logging

logging.basicConfig(level=logging.DEBUG)

def main(event, context):
    # save event to logs
    logging.info(event)

    return {
        'statusCode': 200,
        'body': event
    }
```

As far i can can deduce, it seems almost as if the agent in the layer is not running correctly, but please correct me if i am wrong. Otherwise, please let me know if there is any more information i can provide.
